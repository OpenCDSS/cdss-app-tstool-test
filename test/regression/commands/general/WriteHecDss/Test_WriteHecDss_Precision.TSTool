# Write monthly data to DSS
# Test setup...
StartLog(LogFile="Results/Test_WriteHecDss_Precision.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# RemoveFile(InputFile="Results\Test_WriteHecDss_Precision_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_Precision.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
TS Alias1 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Month.Precision-1",Description="Test data",SetStart="1950-01",SetEnd="1952-03",Units="MGD",PatternValues="5.12345,10.12345,12.579,13.9745,75.38256")
TS Alias2 = NewPatternTimeSeries(NewTSID="Apart2:Bpart2-with-dashes..Cpart2.Month.Precision-2",Description="Test data",SetStart="1950-01",SetEnd="1952-03",Units="MGD",PatternValues="6.12345,11.12345,13.579,16.9745,77.38256")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_Precision_out.dss",TSList=AllMatchingTSID,TSID="Alias1",Precision=1,Type=PER-AVER)
WriteHecDss(OutputFile="Results/Test_WriteHecDss_Precision_out.dss",TSList=AllMatchingTSID,TSID="Alias2",Precision=2,Type=PER-AVER)
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_Precision_out.dss",InputFile="Test_WriteHecDss_Precision.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_Precision.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_Precision.dssutl.out",InputFile2="Results/Test_WriteHecDss_Precision.dssutl.out",WarnIfDifferent=True)
