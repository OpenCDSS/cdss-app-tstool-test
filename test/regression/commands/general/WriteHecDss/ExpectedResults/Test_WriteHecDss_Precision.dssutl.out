  
 /APART1/BPART1-WITH-DASHES/CPART1/01JAN1950/1MON/PRECISION-1/
 Start: 31JAN1950 at 2400 hours;   End: 31MAR1952 at 2400 hours;  Number:   27
 Units: MGD        Type: PER-AVER
 31JAN1950, 2400;      5.1    10.1    12.6    14.0    75.4     5.1    10.1
 31AUG1950, 2400;     12.6    14.0    75.4     5.1    10.1    12.6    14.0
 31MAR1951, 2400;     75.4     5.1    10.1    12.6    14.0    75.4     5.1
 31OCT1951, 2400;     10.1    12.6    14.0    75.4     5.1    10.1
 ----------------------------------------------------------------------------
  
 /APART2/BPART2-WITH-DASHES/CPART2/01JAN1950/1MON/PRECISION-2/
 Start: 31JAN1950 at 2400 hours;   End: 31MAR1952 at 2400 hours;  Number:   27
 Units: MGD        Type: PER-AVER
 31JAN1950, 2400;     6.12   11.12   13.58   16.97   77.38    6.12   11.12
 31AUG1950, 2400;    13.58   16.97   77.38    6.12   11.12   13.58   16.97
 31MAR1951, 2400;    77.38    6.12   11.12   13.58   16.97   77.38    6.12
 31OCT1951, 2400;    11.12   13.58   16.97   77.38    6.12   11.12
 ----------------------------------------------------------------------------
  
